addons/delete-others/userscript.js

```diff
@@ -0,0 +1,64 @@
+export default async function ({ addon, console, msg }) {
+  const vm = await addon.tab.traps.vm;
+
+  const types = ["sound", "costume"];
+  let deletedItems = [];
+  let deleted;
+  let target;
+
+  function getRestoreFun(type) {
+    if (type === "costume") {
+      deletedItems.forEach((costume) => {
+        vm.addCostume(costume.md5, costume, target.id, "");
+      });
+    } else if (type === "sound") {
+      deletedItems.forEach((sound) => {
+        vm.addSound(sound, target.id);
+      });
+    }
+    deletedItems = [];
+  }
+
+  addon.tab.createEditorContextMenu(
+    (ctx) => {
+      if (window.confirm(`Are you sure you want to delete all other ${ctx.type}s?`)) {
+        const type = ctx.type === "costume" ? "Costume" : "Sound";
+
+        deletedItems = [];
+        target = vm.editingTarget;
+
+        const loop = ctx.type === "costume" ? target.getCostumes().length - 1 : target.getSounds().length - 1;
```

Comment:
"loop" is a strange naming choice.

Reply:
I felt it was easier than "numberOfAssets" imo it's fine

Reply:
If you need me to, I'll change it, it's not like I've typed a full variable name in vsc after its declared